packages/cli/command_help.pony

```diff
@@ -0,0 +1,196 @@
+
+primitive Help
+
+  fun general(cs: CommandSpec box): CommandHelp
+  =>
+    CommandHelp._create(None, cs)
+
+  fun for_command(cs: CommandSpec box, argv: Array[String] box):
+    (CommandHelp | SyntaxError)
+  =>
+    let ch = CommandHelp._create(None, cs)
+    _parse(cs, ch, argv)
```

Comment:
Since both of these methods require a `CommandSpec box` as the first argument, it seems like they could be moved to be a `fun box` on the `CommandSpec` type.

Then we could eliminate the need for the `Help` type, or at least make it private if you wanted to keep the `_parse` method here.